src/server/tablet_snap.rs

```diff
@@ -80,21 +84,57 @@ fn is_sst(file_name: &str) -> bool {
     file_name.ends_with(".sst")
 }
 
-async fn read_to(f: &mut File, to: &mut Vec<u8>, size: usize, limiter: &Limiter) -> Result<()> {
```

Comment:
You don't need to introduce `WrappedReadableFile`, just changing `&mut File` to `&mut impl Read` should work.

Reply:
To accomplish that, you will need to abstract all code interacting with files into a generic (like `is_sst_match_preview(f: impl Read)`, not just this one function. It is a bigger change than the current way.

Reply:
Even extracting the single function `read_to` is good for readability. When seeing `WrappedReadableFile` is constructed before sending, it seems the file is decrypted before sending. But turns out it's not as `None` stands for no decryption. If `read_to` doesn't depend on `WrappedReadableFile`, `File` can be used directly, hence no misunderstanding.